esp-hal/src/gpio.rs

```diff
@@ -1854,6 +1863,29 @@ impl IO {
             pins,
         }
     }
+
+    /// Install the given interrupt handler replacing any previously set
+    /// handler.
+    ///
+    /// When the async feature is enabled the handler will be called first and
+    /// the internal async handler will run after. In that case it's
+    /// important to not reset the interrupt status when mixing sync and
+    /// async (i.e. using async wait) interrupt handling.
+    pub fn set_interrupt_handler(&mut self, handler: unsafe extern "C" fn() -> ()) {
+        critical_section::with(|_cs| {
+            #[cfg(feature = "async")]
```

Comment:
I think we want to avoid cfging with async as we might get rid of it. Imo we can always set this user handler, I hope the cost of calling an extra function inside GPIO is not too much.

Reply:
Good idea - makes things much nicer and easier